xml-reader/src/main/java/org/triplea/generic/xml/reader/XmlMapper.java

```diff
@@ -0,0 +1,70 @@
+package org.triplea.generic.xml.reader;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.XMLStreamReader;
+import lombok.AllArgsConstructor;
+
+@AllArgsConstructor
+public class XmlMapper {
+  private XMLStreamReader xmlStreamReader;
+
+  public <T> T mapXmlToClass(final Class<T> pojo) throws XMLStreamException {
+    // Todo: do validation on which kinds of tags and types we have.
+    // for example, a '@Tag' should not be on a list.
+
+    try {
+      final Constructor<T> constructor = pojo.getDeclaredConstructor();
+      constructor.setAccessible(true);
+      final T instance = constructor.newInstance();
+
+      final XmlParser.TagParser tagParser = new XmlParser.TagParser(pojo.getSimpleName());
+      boolean hasNestedTags = false;
+
+      for (final Field field : pojo.getDeclaredFields()) {
+        if (field.getAnnotation(Attribute.class) != null) {
+          field.setAccessible(true);
+
+          final String value =
+              Optional.ofNullable(xmlStreamReader.getAttributeValue(null, field.getName()))
+                  .orElseGet(() -> field.getAnnotation(Attribute.class).defaultValue());
+          field.set(instance, value);
+        }
+        if (field.getAnnotation(Tag.class) != null) {
+          field.setAccessible(true);
+          tagParser.childTagHandler(
+              field.getName(), () -> field.set(instance, mapXmlToClass(field.getType())));
+          hasNestedTags = true;
+        } else if (field.getAnnotation(TagList.class) != null) {
+          field.setAccessible(true);
+          if (field.getType() == List.class) {
+            final List tagList = new ArrayList();
```

Comment:
Ahhh raw types 😰 
```suggestion
            final List<Object> tagList = new ArrayList<>();
```

Another potential approach would be to make the tagParser thingy part of a dedicated method that uses a generic parameter something like this, which makes it clear that the list and the field are supposed to be the same generic type, but doesn't make the code more readable IMO.

```java
@SuppressWarnings("unchecked")
<T> List<?> populateList() throws Exception {
  final List<T> tagList = new ArrayList<>();
  final Class<T> listType = (Class<T>) field.getAnnotation(TagList.class).value();
  tagParser.childTagHandler(listType.getSimpleName(), () -> tagList.add(mapXmlToClass(listType)));
  return tagList;
}
```

Reply:
I'd wonder if the class cast might cause issues based on the XML data and make error messaging difficult at all. To another extent, raw typing is going to be somewhat unavoidable. Not a bad suggestion though. I'm about ready to do some polishing of the XML-reader module and will see if I can incorporate this.

Reply:
I mean the first example should work too and gets rid of the compiler warning